rules/CodeQuality/NodeAnalyzer/ConstructorPropertyDefaultExprResolver.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\CodeQuality\NodeAnalyzer;
+
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\Array_;
+use PhpParser\Node\Expr\ArrayItem;
+use PhpParser\Node\Expr\Assign;
+use PhpParser\Node\Expr\ConstFetch;
+use PhpParser\Node\Expr\PropertyFetch;
+use PhpParser\Node\Scalar;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\Stmt\Expression;
+use Rector\CodeQuality\ValueObject\DefaultPropertyExprAssign;
+use Rector\NodeNameResolver\NodeNameResolver;
+
+final class ConstructorPropertyDefaultExprResolver
+{
+    public function __construct(
+        private readonly NodeNameResolver $nodeNameResolver
+    ) {
+    }
+
+    /**
+     * @return DefaultPropertyExprAssign[]
+     */
+    public function resolve(ClassMethod $classMethod): array
+    {
+        $stmts = $classMethod->getStmts();
+        if ($stmts === null) {
+            return [];
+        }
+
+        $defaultPropertyExprAssigns = [];
+
+        foreach ($stmts as $stmt) {
+            if (! $stmt instanceof Expression) {
+                continue;
+            }
+
+            $nestedStmt = $stmt->expr;
+            if (! $nestedStmt instanceof Assign) {
+                continue;
+            }
+
+            $assign = $nestedStmt;
+
+            if (! $assign->var instanceof PropertyFetch) {
+                continue;
+            }
+
+            $propertyFetch = $assign->var;
+            if (! $this->nodeNameResolver->isName($propertyFetch->var, 'this')) {
+                continue;
+            }
+
+            $propertyName = $this->nodeNameResolver->getName($propertyFetch->name);
+            if (! is_string($propertyName)) {
+                continue;
+            }
+
+            $expr = $assign->expr;
+            if (! $this->isAllowedPropertyDefaultExpr($expr)) {
+                continue;
+            }
+
+            $defaultPropertyExprAssigns[] = new DefaultPropertyExprAssign($stmt, $propertyName, $expr);
+        }
+
+        return $defaultPropertyExprAssigns;
+    }
+
+    private function isAllowedPropertyDefaultExpr(Expr $expr): bool
```

Comment:
There is already service !ExprAnalyzer->isDynamicValue() for it

https://github.com/rectorphp/rector-src/blob/a2422d7937a9859a403b52691f99617716e6e15a/src/NodeAnalyzer/ExprAnalyzer.php#L67

Reply:
I'll try, thanks